src/middlewared/middlewared/plugins/gluster_linux/peer.py

```diff
@@ -35,161 +41,102 @@ def _parse_peer(self, p):
-
+    async def _parse_peer_status_xml(self, data):
         peers = []
         for _ in data.findall('peerStatus/peer'):
             try:
-                peers.append(self._parse_peer(_))
+                peers.append(await self._parse_peer(_))
             except Exception as e:
                 raise CallError(
                     f'Failed parsing peer information with error: {e}'
                 )
 
         return peers
 
-    @private
-    async def resolve_host_or_ip(self, hostname, verrors):
-
-        args = (self.middleware, verrors, 'resolve_host_or_ip', hostname)
-        return await resolve_hostname(*args)
-
-    @private
-    def common_validation(self, hostname=None):
-
-        verrors = ValidationErrors()
-
-        if hostname:
-            self.middleware.call_sync(
-                'gluster.peer.resolve_host_or_ip', hostname, verrors)
-
-        verrors.check()
-
-    @accepts(
-        Dict(
-            'probe_peer_create',
-            Str('hostname', required=True, max_length=253)
-        )
-    )
+    @accepts(Dict(
+        'peer_create',
+        Str('hostname', required=True, max_length=253)
+    ))
     @job(lock=GLUSTER_JOB_LOCK)
-    def create(self, job, data):
+    async def do_create(self, job, data):
         """
         Add peer to the Trusted Storage Pool.
 
-        `hostname` can be an IP(v4/v6) address or DNS name.
+        `hostname` String representing an IP(v4/v6) address or DNS name
         """
 
-        hostname = data.get('hostname')
-
-        self.common_validation(hostname=hostname)
+        await self.middleware.call('gluster.method.run', peer.attach, {'args': (data['hostname'],)})
+        return await self.middleware.call('gluster.peer.query', [('hostname', '=', data['hostname'])])
 
-        return run_method(peer.attach, hostname)
-
-    @accepts(
-        Dict(
-            'probe_peer_delete',
-            Str('hostname', required=True, max_length=253)
-        )
-    )
+    @accepts(Str('id'))
     @job(lock=GLUSTER_JOB_LOCK)
-    def delete(self, job, data):
-        """
-        Remove peer of `hostname` from the Trusted Storage Pool.
+    async def do_delete(self, job, id):
         """
+        Remove peer of `id` from the Trusted Storage Pool.
 
-        hostname = data.get('hostname')
-
-        self.common_validation(hostname=hostname)
-
-        return run_method(peer.detach, hostname)
-
-    @accepts()
-    def status(self):
-        """
-        List the status of peers in the Trusted Storage Pool
-        excluding localhost.
+        `id` String representing the uuid of the peer
         """
 
-        return run_method(peer.status)
+        await self.middleware.call(
+            'gluster.method.run', peer.detach, {'args': ((await self.get_instance(id))['hostname'],)})
 
-    @accepts()
-    def pool(self):
+    @accepts(Dict(
+        'peer_status',
+        Bool('localhost', default=True),
+    ))
+    async def status(self, data):
```

Comment:
We seem to be doing xml things here like parsing and finding. I wonder if it would make sense to keep it in a method and not a coroutine so the event loop is not potentially blocked?